### PR TITLE
Support depth option for memory adapter

### PR DESCRIPTION
**Pull request by 50kudos**
This pull request completes https://github.com/florinpatrascu/closure_table/issues/1 since we already shipped the first part.

### CODE REVIEW

File: lib/cte/adapter/memory.ex [hunk 1]
```diff
@@ -127,72 +141,121 @@ defmodule CTE.Adapter.Memory do
 
   @doc false
   defp _descendants(ancestor, opts, config) do
-    fn path, {descendants, size} = acc, itself? ->
+    descendants_collector(ancestor)
+    |> _find_leaves(opts, config)
+    |> depth(opts, config)
+    |> selected(opts, config)
```
Comment (florinpatrascu) [inline lib/cte/adapter/memory.ex]: maybe we can extract this into a separate function, and reuse it for _ancestors as well? But the whole work looks very good, thank you!

Comment (50kudos) [reply lib/cte/adapter/memory.ex]: What I did did not have clean api yet, so thank you for feedback!
You meant all these 3 functions, right? Feel free to suggest what will make it cleaner.

File: lib/cte/adapter/memory.ex [hunk 2]
```diff
@@ -127,72 +141,121 @@ defmodule CTE.Adapter.Memory do
...
     %CTE{nodes: nodes, paths: paths} = config
 
     with true <- Map.has_key?(nodes, ancestor) do
-      leaf_new_ancestors =
-        _ancestors(ancestor, opts, config)
-        |> Enum.map(&[&1, leaf])
+      {leaf_new_ancestors, _} =
+        ancestors_collector(ancestor)
+        |> _find_leaves([{:itself, true} | opts], config)
+        |> depth(opts, config)
```
Comment (50kudos) [inline lib/cte/adapter/memory.ex]: I think I forgot the remove this. And all `opts` should not be used at all for `_insert/4`?

Comment (florinpatrascu) [reply lib/cte/adapter/memory.ex]: true

**Comment by florinpatrascu** [comment]
💯

**Pull request merged**